src/main/java/com/kmbl/InventoryManagementService/services/InventoryService.java

```diff
@@ -77,4 +77,52 @@ public void freeInventory(CancelOrderMessage message) throws ServiceException {
     public void deleteInventory(String id) {
         inventoryRepository.deleteById(id);
     }
+
+
+    public ResponseBody updateInventoryforOrder(List<OrderRequestBody> orderRequestItems) {
+        synchronized (this) {
+            InventoryOrderStatus overallStatus = InventoryOrderStatus.FAILED_ORDER;
+            ResponseBody response = new ResponseBody();
+            for (int i = 0; i < orderRequestItems.size(); i++) {
+                Optional<Inventory> tempInventory = inventoryRepository.findBySellerIdAndProductId(orderRequestItems.get(i).getSellerID(),orderRequestItems.get(i).getProductID());
+                Inventory inventoryFinal = null;
+                if(tempInventory.isEmpty()){
+                    InventoryOrderItem tempInventoryOrderItem =new InventoryOrderItem(orderRequestItems.get(i).getProductID(), orderRequestItems.get(i).getSellerID(), String.valueOf(orderRequestItems.get(i).getCount()));
```

Comment:
Why sending Count as String, Secondly if inventory is empty you should return count as zero(0) rather than the item count.